dags/process_elife_zip_dag.py

```diff
@@ -7,20 +7,25 @@
 from io import BytesIO
 from pathlib import Path
 from tempfile import TemporaryFile
+from xml.dom import XML_NAMESPACE
 from zipfile import ZipFile
 
 from airflow import DAG, configuration
 from airflow.operators import python_operator
 from airflow.utils import timezone
 from lxml import etree
+from lxml.builder import ElementMaker
 
-from aws import get_aws_connection, list_bucket_keys_iter
+from aws import get_aws_connection
 from task_helpers import get_previous_task_name
 
 ALLOWED_EXTENSIONS = {'.jpg', '.jpeg', '.pdf', '.xml'}
 
+ENDPOINT_URL = configuration.conf.get('libero', 'endpoint_url') or 'https://s3.amazonaws.com'
```

Comment:
I think the naming of this configuration setting can be improved as `endpoint` in AWS jargon is typically the target of an API call done by the SDK (which follows a XML-based format with signatures, etc.)
What we are after here is a more abstract URL prefix where the images (or other assets) will be accessible. In theory, you could even put a proxy like a gateway that authenticates or caches the files in between. I would configure the whole `EXPANDED_URL` including the bucket name.

Reply:
Moreover, I just discovered to be [future proof](https://aws.amazon.com/blogs/aws/amazon-s3-path-deprecation-plan-the-rest-of-the-story/) buckets should be accessed via a `https://${bucket-name}.s3.amazonaws.com` URL.

Reply:
This is used here: https://github.com/libero/elife-style-content-adapter-prototype/blob/master/dags/aws.py#L7

I could rename it to `boto_endpoint_url`?